crates/bevy_log/src/lib.rs

```diff
@@ -173,7 +173,11 @@ impl Plugin for LogPlugin {
             #[cfg(feature = "tracing-tracy")]
             let tracy_layer = tracing_tracy::TracyLayer::new();
 
-            let fmt_layer = tracing_subscriber::fmt::Layer::default().with_writer(std::io::stderr);
+            let use_color = std::env::var("NO_COLOR").map(|v| v != "1").unwrap_or(true);
```

Comment:
If the env variable is present, color should not be used regardless of its value.

Reply:
I initially thought the same thing. But if you carefully read the nocolor website, command line flags are supposed to override the environment variable. Because of this, the upstream change allows `with_ansi()` to override the environment variable, otherwise it wouldn’t be possible for command-line flags to override the environment variable. 

Longer explanation in this comment in the upstream issue for the PR https://github.com/tokio-rs/tracing/issues/2388#issuecomment-1633380824

Reply:
Bah, they literally just updated the nocolor website a week ago, and the text about flag overriding has been removed. 

If you want to push for NO_COLOR to always override commandline flags, I’d open a new PR upstream to fix the behavior there. The reason can be that the NO_COLOR guidelines changed.

Reply:
I found it in the code block on the no-color website. It’s the comment at the bottom of their example code that says flags can override environment. 

```c

int
main(int argc, char *argv[])
{
	char *no_color = getenv("NO_COLOR");
	bool color = true;

	if (no_color != NULL && no_color[0] != '\0')
		color = false;

	/* do getopt(3) and/or config-file parsing to possibly turn color back on */
	...
}
```

Reply:
from my understanding the check should be on "not empty". if it's set but empty, keep the colours, if it's set and not empty, then remove the colours

Reply:
I remember reading the website a while back and wondering why they didn't say `NO_COLOR="true"`